piptools/repositories/pypi.py

```diff
@@ -227,6 +229,23 @@ def get_dependencies(self, ireq):
 
         return self._dependencies_cache[ireq]
 
+    def _get_project(self, ireq):
+        """
+        Given an `InstallRequirement`, return a dict of a project info
+        from PyPI JSON API. Return None on HTTP or JSON parsing error.
+
+        See https://warehouse.readthedocs.io/api-reference/json/
+        """
+        url = "{url}/{name}/json".format(url=PyPI.pypi_url, name=ireq.name)
```

Comment:
A thought: probably, we should configure a `PyPI.pypi_url`.

Reply:
Yeah, this part must be improved when a user has an extra URLs, or a private index url, or ignores package index by `--no-index`.

Reply:
@graingert any thoughts on this?

Reply:
I'm not sure about this, we could try to see if the index chosen supports the PyPI JSON API, or use hashes from PyPI.org for files with the same filename, or ignore it when using a non PyPI.org index

or make it a cli flag?

Reply:
Addressed in ed6d8158d3aa0a161c2ec4403677dcac4515f087.

This handles all configured index URLs and also handles disabled package index via `--no-index`.